Mathlib/Init/Data/Fin/Basic.lean

```diff
@@ -1,34 +0,0 @@
-/-
-Copyright (c) 2016 Microsoft Corporation. All rights reserved.
-Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Leonardo de Moura
--/
-import Mathlib.Data.Nat.Notation
-
-/-!
-# Note about `Mathlib/Init/`
-The files in `Mathlib/Init` are leftovers from the port from Mathlib3.
-(They contain content moved from lean3 itself that Mathlib needed but was not moved to lean4.)
-
-We intend to move all the content of these files out into the main `Mathlib` directory structure.
-Contributions assisting with this are appreciated.
-
-# Theorems about equality in `Fin`.
--/
-
-namespace Fin
-
-variable {n : ℕ} {i j : Fin n}
-
-@[deprecated eq_of_val_eq (since := "2024-02-15")]
-theorem eq_of_veq : i.val = j.val → i = j := eq_of_val_eq
-
-@[deprecated val_eq_of_eq (since := "2024-02-15")]
-theorem veq_of_eq : i = j → i.val = j.val := val_eq_of_eq
-
--- These two aren't deprecated because `ne_of_val_ne` and `val_ne_of_ne`
--- use `¬a = b` instead of `a ≠ b`. TODO: fix or rename in Lean core.
```

Comment:
This comment has not been addressed. But seemingly this syntactic difference does not cause any problems in mathlib.

Reply:
Can we open a PR to core that changes these two lemmas to use `≠`? That seems uncontroversial to me.

Reply:
@eric-wieser https://github.com/leanprover/lean4/pull/5011

Reply:
I've merged this now.